app/components/base_component.rb

```diff
@@ -1,17 +1,35 @@
 class BaseComponent < ViewComponent::Base
   def before_render
-    return if @rendered_scripts
-    @rendered_scripts = true
-    if helpers.respond_to?(:enqueue_component_scripts) && self.class.scripts.present?
-      helpers.enqueue_component_scripts(*self.class.scripts)
-    end
+    render_assets unless @has_rendered_assets
   end
 
   def self.scripts
-    @scripts ||= _sidecar_files(['js', 'ts']).map { |file| File.basename(file, '.*') }
+    @scripts ||= sidecar_files_basenames(['js', 'ts'])
+  end
+
+  def self.stylesheets
+    @stylesheets ||= sidecar_files_basenames(['scss'])
   end
 
   def unique_id
     @unique_id ||= SecureRandom.hex(4)
   end
+
+  private
+
+  def self.sidecar_files_basenames(extensions)
+    _sidecar_files(extensions).map { |file| File.basename(file, '.*') }
+  end
+
+  def render_assets
+    if helpers.respond_to?(:enqueue_component_scripts) && self.class.scripts.present?
```

Comment:
I see it was in the old code, but can you remind me why we need to do a `respond_to?` check for that method?

Reply:
> I see it was in the old code, but can you remind me why we need to do a `respond_to?` check for that method?

A bit of speculative thinking that these components could be used outside Rails, based on https://github.com/18F/identity-idp/pull/5398#issuecomment-923026492 , where `enqueue_component_scripts` would be optionally provided by the framework as a way to control how scripts (and now styles) are printed.